js/ui/button_group.js

```diff
@@ -0,0 +1,252 @@
+import $ from "../core/renderer";
+import Widget from "./widget/ui.widget";
+import Button from "./button";
+import CollectionWidget from "./collection/ui.collection_widget.edit";
+import registerComponent from "../core/component_registrator";
+import DataExpressionMixin from "./editor/ui.data_expression";
+import { extend } from "../core/utils/extend";
+import BindableTemplate from "./widget/bindable_template";
+import { isPlainObject } from "../core/utils/type";
+
+const BUTTON_GROUP_CLASS = "dx-buttongroup",
+    BUTTON_GROUP_WRAPPER_CLASS = "dx-buttongroup-wrapper";
+
+var ButtonCollection = CollectionWidget.inherit({
+    _initMarkup() {
+        this._updateSelectedItemKeysByIndexes(this.option("selectedIndexes"));
+        this.callBase();
+    },
+
+    _getDefaultOptions() {
+        return extend(this.callBase(), DataExpressionMixin._dataExpressionDefaultOptions(), {
+            selectedIndexes: []
+        });
+    },
+
+    _renderItemContent(options) {
+        options.container = $(options.container).parent();
+        this.callBase(options);
+    },
+
+    _focusTarget() {
+        return this.$element().parent();
+    },
+
+    _keyboardEventBindingTarget() {
+        return this._focusTarget();
+    },
+
+    _refreshContent() {
+        this._prepareContent();
+        this._renderContent();
+    },
+
+    _updateSelectedItemKeysByIndexes(indexes) {
+        if(!indexes.length) {
+            return;
+        }
+
+        const selectedItemKeys = indexes.map((index) => this._getKeyByIndex(index));
+        this.option("selectedItemKeys", selectedItemKeys);
+    },
+
+    _optionChanged(args) {
+        this.callBase(args);
+
+        if(this._cancelOptionChange === args.name) {
+            return;
+        }
+
+        switch(args.name) {
+            case "selectedIndexes":
+                this._updateSelectedItemKeysByIndexes(args.value);
+                break;
+            case "selectedItemKeys":
+                this._setOptionSilent("selectedIndexes", this._getSelectedItemIndices(args.value));
+                break;
+        }
+    }
+});
+
+/**
+ * @name dxButtonGroup
+ * @inherits Widget
+ * @hasTranscludedContent
+ * @module ui/button_group
+ * @export default
+ */
+const ButtonGroup = Widget.inherit({
```

Comment:
Not CollectionWidget?

Reply:
How and the RadioGroup widget. The options of Collection widget not needed in the ButtonGroup but excluding the items and selectedItemKeys.

Reply:
RadioGroup is an editor - it has the `value` option and didn't support an item selection.
We can hide the part of the options

Reply:
@lukyanovas  told me that is a solution is bad.